### PR TITLE
Remove ARM32 Support

### PR DESCRIPTION
**Pull request by NickGerleman**
This PR attempts a full removal of ARM32 support from our codebase (including build logic, NuGet packages, etc).

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7645)

### CODE REVIEW

File: packages/e2e-test-app/windows/ReactUWPTestApp.sln [hunk 1]
```diff
@@ -46,52 +46,36 @@ Global
 		..\..\..\vnext\Shared\Shared.vcxitems*{f7d32bd0-2749-483e-9a0d-1635ef7e3136}*SharedItemsImports = 4
 	EndGlobalSection
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
-		Debug|ARM = Debug|ARM
-		Debug|ARM64 = Debug|ARM64
```
Comment (dannyvv) [inline packages/e2e-test-app/windows/ReactUWPTestApp.sln:50]: note: you are removing Arm64 as well... Maybe regex bug?

Comment (dannyvv) [reply packages/e2e-test-app/windows/ReactUWPTestApp.sln:50]: note: multiple removals of arm64 in this file...

**Review by dannyvv — APPROVED** [review]
Thanks!!! 😃

**Comment by dannyvv** [comment]
Few missed places:
* Comment in: https://github.com/microsoft/react-native-windows/blob/2c5ba22a3cfa4447dd227b4f7274c882e1786fed/packages/%40react-native-windows/cli/powershell/Add-AppDevPackage.ps1#L397
* No need for ARM Cpp deps: Search for: `$WindowsSDKOptions`
* Pre-and-Pack-Nuget: Search for `- ${{ if eq(parameters.packMicrosoftReactNative, true) }}:` and `Copy the x64/release ref assembly to anycpu folder`

File: change/react-native-windows-8659aa8e-46be-4ef9-96ce-b3d5d82bc5ad.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "☠ ARM32",
```
Comment (asklar) [inline change/react-native-windows-8659aa8e-46be-4ef9-96ce-b3d5d82bc5ad.json:3]: Can you replace this with something human readable like "Remove ARM32 support" ?
Same for the PR title so people can find it in searches in the future

File: packages/e2e-test-app/windows/ReactUWPTestApp.sln (hunk 1 above)
Comment (asklar) [reply packages/e2e-test-app/windows/ReactUWPTestApp.sln:50]: Good catch

File: packages/sample-apps/windows/SampleApps.sln [hunk 3]
```diff
@@ -101,10 +88,6 @@ Global
 		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Debug|x86.ActiveCfg = Debug|Win32
 		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Debug|x86.Build.0 = Debug|Win32
 		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Debug|x86.Deploy.0 = Debug|Win32
-		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Release|ARM.ActiveCfg = Release|ARM
-		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Release|ARM.Build.0 = Release|ARM
-		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Release|ARM.Deploy.0 = Release|ARM
-		{93F7572C-64B9-4096-9EF9-6BA0EDE2B50D}.Release|ARM64.ActiveCfg = Release|ARM64
```
Comment (asklar) [inline packages/sample-apps/windows/SampleApps.sln:107]: Arm64 here too

File: vnext/Microsoft.ReactNative.sln [hunk 4]
```diff
@@ -139,198 +135,152 @@ Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|x64 = Debug|x64
 		Debug|x86 = Debug|x86
-		Debug|ARM = Debug|ARM
 		Debug|ARM64 = Debug|ARM64
 		Release|x64 = Release|x64
 		Release|x86 = Release|x86
-		Release|ARM = Release|ARM
-		Release|ARM64 = Release|ARM64
```
Comment (asklar) [inline vnext/Microsoft.ReactNative.sln:147]: Arm64

File: vnext/Microsoft.ReactNative.sln [hunk 5]
```diff
@@ -139,198 +135,152 @@ Global
...
-		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|ARM.ActiveCfg = Debug|ARM
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|ARM64.ActiveCfg = Debug|ARM64
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x64.ActiveCfg = Debug|x64
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x64.Build.0 = Debug|x64
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x64.Deploy.0 = Debug|x64
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x86.ActiveCfg = Debug|x86
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x86.Build.0 = Debug|x86
 		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Debug|x86.Deploy.0 = Debug|x86
-		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Release|ARM.ActiveCfg = Release|ARM
-		{46D76F7A-8FD9-4A7D-8102-2857E5DA6B84}.Release|ARM64.ActiveCfg = Release|ARM64
```
Comment (asklar) [inline vnext/Microsoft.ReactNative.sln:244]: Arm64

**Review by asklar — CHANGES_REQUESTED** [review]

**Comment by NickGerleman** [comment]
> Few missed places:
> 
> * Comment in: https://github.com/microsoft/react-native-windows/blob/2c5ba22a3cfa4447dd227b4f7274c882e1786fed/packages/%40react-native-windows/cli/powershell/Add-AppDevPackage.ps1#L397
> * No need for ARM Cpp deps: Search for: https://github.com/microsoft/react-native-windows/blob/2c5ba22a3cfa4447dd227b4f7274c882e1786fed/vnext/Scripts/Install-WindowsSDKISO.ps1#L12
> * Pre-and-Pack-Nuget: Search for https://github.com/microsoft/react-native-windows/blob/2c5ba22a3cfa4447dd227b4f7274c882e1786fed/.ado/templates/prep-and-pack-nuget.yml#L41
>    and https://github.com/microsoft/react-native-windows/blob/2c5ba22a3cfa4447dd227b4f7274c882e1786fed/.ado/templates/prep-and-pack-nuget.yml#L41

The top script here is a copy from somewhere else, so will leave that one as-is. Will fix up the others.

**Review by asklar — APPROVED** [review]

**Pull request merged**